Grievance_Backend/grievance/src/main/java/com/grievance/service/TicketServiceImpl.java

```
@@ -60,6 +73,56 @@ public Optional<List<TicketOutDto>> listOfAllTickets() {
     return Optional.ofNullable(tickets);
   }
 
+  /**
+   * method to access rickets by their department Name.
+   * @param departmentName
+   * @return list of ticket out DTO
+   */
+  @Override
+  public Optional<List<TicketOutDto>> listOfAllTicketsByDepartmentName(
+    final String departmentName
```

Don't return Optional , just return OutDto also add the check before returning , if the Object is null you can throw an exception RecordNotFoundException.

Make this changes in all methods.

---

code is updated.